l2geth/core/genesis.go

```diff
@@ -299,22 +299,13 @@ func ApplyOvmStateToState(statedb *state.StateDB, stateDump *dump.OvmDump, l1XDo
 		l1MessengerValue := common.BytesToHash(l1XDomainMessengerAddress.Bytes())
 		statedb.SetState(AddressManager.Address, l1MessengerSlot, l1MessengerValue)
 	}
-	OVM_ETH, ok := stateDump.Accounts["OVM_ETH"]
+	OVM_L2StandardBridge, ok := stateDump.Accounts["OVM_L2StandardBridge"]
 	if ok {
-		log.Info("Setting OVM_L1ETHGateway in OVM_ETH", "address", l1ETHGatewayAddress.Hex())
-		if strings.Contains(OVM_ETH.Code, "a84ce98") {
-			// Set the gateway of OVM_ETH at new dump
-			log.Info("Detected current OVM_ETH dump, setting slot 0x1 ")
-			l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000001")
-			l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
-			statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
-		} else {
-			// Set the gateway of OVM_ETH at legacy slot
-			log.Info("Detected legacy OVM_ETH dump, setting slot 0x8")
-			l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000008")
-			l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
-			statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
-		}
+		log.Info("Setting OVM_L1StandardBridge in OVM_L2StandardBridge", "address", l1StandardBridgeAddress.Hex())
+		// Set the gateway of OVM_L2StandardBridge at new dump
+		l1BridgeSlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000001")
```

Comment:
What kind of safeguard do you have in mind? Does that mean writing an integration test that reads the value at this slot and compares it to the value returned by calling the getter on the contract?

Reply:
Yes I think a test in the JS and if the JS test breaks then we know we need to update this. It may be possible to use `core.ApplyMessage` in the future so that we do not need to worry about this anymore

Reply:
Did this in the deployment script.